src/update_helper.rs

```diff
@@ -17,12 +17,26 @@ impl Context {
         scope: Param,
         new_timestamp: i64,
     ) -> Result<bool> {
-        let mut contact = Contact::load_from_db(self, contact_id).await?;
-        if contact.param.update_timestamp(scope, new_timestamp)? {
-            contact.update_param(self).await?;
-            return Ok(true);
-        }
-        Ok(false)
+        self.sql
+            .transaction(|transaction| {
+                let mut param: Params = transaction.query_row(
+                    "SELECT param FROM contacts WHERE id=?",
+                    [contact_id],
+                    |row| {
+                        let param: String = row.get(0)?;
+                        Ok(param.parse().unwrap_or_default())
+                    },
+                )?;
+                let update = param.update_timestamp(scope, new_timestamp)?;
+                if update {
+                    transaction.execute(
+                        "UPDATE contacts SET param=? WHERE id=?",
```

Comment:
this is not covered by any test, eg. of you change this line to `"UPDATE contacts SET param=? WHERE id=? AND id=123456`, no test will fail.

no regression and no blocker, this was not better before. however, i will do a pr with a test i created during review.

Reply:
added a test in https://github.com/deltachat/deltachat-core-rust/pull/4090